ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -167,8 +167,13 @@ private void runAgent(MLAgent mlAgent, Map<String, String> params, ActionListene
                     executeParams.put(key.replace(toolSpec.getType() + ".", ""), params.get(key));
                 }
             }
-            log.info("Fetching tool for type: " + toolSpec.getType());
-            Tool tool = toolFactories.get(toolSpec.getType()).create(executeParams);
+            String toolType = toolSpec.getType();
+            log.info("Fetching tool for type: " + toolType);
+
+            if (toolFactories.get(toolType) == null) {
+                log.error(toolType + " can't be found, please use ListTool API to check if it's deployed.");
```

Comment:
Need suggestions how we should exit gracefully if the tool suggested by LLM is not available.

Reply:
@arjunkumargiri do you have any suggestion here?

Reply:
@arjunkumargiri, please take a look at here, it's important one.

Reply:
In addition to logging, agent should respond back to LLM that the suggested tool is not available and to either use a different tool from the available list of tool or to generate response from the available data.

@jngz-es please let me know if you have any concerns.